engine-tests/src/test/java/org/terasology/documentation/apiScraper/ApiComparator.java

```diff
@@ -0,0 +1,289 @@
+     * @param br BufferedReader containing an api file content
+     * @return A map with the api classes and interfaces as keys;
+     * Their methods and as a list of ApiMethods in the values
+     * @throws Exception if the readLine fails.
+     */
+    private static Map<String, Collection<ApiMethod>> getApi(BufferedReader br) throws Exception {
+        String line = br.readLine();
+        Map<String, Collection<ApiMethod>> api = new HashMap<>();
+        while (line != null) {
+            if (line.startsWith("*")) {
+                if (line.endsWith("(PACKAGE)")) {
+                    line = br.readLine();
+                    continue;
+                }
+                String className = line;
+                String aux;
+                api.put(className, new ArrayList<>());
+                ApiMethod method;
+                aux = br.readLine();
+                while ((aux != null && (aux.endsWith("(METHOD)")
+                        || aux.endsWith("(CONSTRUCTOR)")
+                        || aux.endsWith("(ABSTRACT METHOD)")))) {
+
+                    //Checks if its a method or constructor
+                    if (aux.endsWith("(METHOD)") || aux.endsWith("(ABSTRACT METHOD)")) {
+                        String returnType = br.readLine();
+                        String parameters = br.readLine();
+                        String exceptionType = br.readLine();
+                        method = new ApiMethod(className, aux, returnType, exceptionType, parameters);
+                    } else {
+                        String returnType = "";
+                        String parameters = br.readLine();
+                        String exceptionType = "";
+                        method = new ApiMethod(className, aux, returnType, exceptionType, parameters);
+                    }
+
+
+                    api.get(className).add(method);
+                    aux = br.readLine();
+                }
+                line = aux;
+
+            } else {
+                line = br.readLine();
+            }
+        }
+        return api;
+    }
+
+    private static void checkClassAdditionAndDeletion(Map<String, Collection<ApiMethod>> originalApi, Map<String, Collection<ApiMethod>> newApi) {
+        System.out.println("Checking Class Addition and Deletion");
+        for (String className : originalApi.keySet()) {
+            if (!newApi.containsKey(className)) {
+                System.out.println("MAJOR INCREASE, DELETION OF " + className);
+            }
+        }
+
+        for (String className : newApi.keySet()) {
+            if (!originalApi.containsKey(className)) {
+                System.out.println("MINOR INCREASE, ADDITION OF " + className);
+            }
+        }
+
+    }
+
+    /**
+     * Checks creation and deletion of methods, as well as existing method changes
+     * @param originalApi the original api generated from "API_file.txt"
```

Comment:
You might want to move the "API_file.txt" and related to constants and refer to those instead